bin/runtime/src/chain_extension/tests/executor.rs

```diff
@@ -0,0 +1,75 @@
+use pallet_snarcos::{Error as SnarcosError, ProvingSystem, VerificationKeyIdentifier};
+
+use crate::chain_extension::executor::Executor;
+
+type Error = SnarcosError<()>;
+type Result = core::result::Result<(), Error>;
+
+/// Describes how the `Executor` should behave when one of its methods is called.
+#[derive(Clone, Eq, PartialEq)]
+pub(super) enum Responder {
+    /// Twist and shout.
+    Panicker,
+    /// Return `Ok(())`.
+    Okayer,
+    /// Return `Err(Error)`.
+    Errorer(Error),
+}
+
+/// Testing counterpart for `Runtime`.
+///
+/// `STORE_KEY_RESPONDER` instructs how to behave then `store_key` is called.
+/// `VERIFY_RESPONDER` instructs how to behave then `verify` is called.
+pub(super) struct MockedExecutor<
+    const STORE_KEY_RESPONDER: Responder,
+    const VERIFY_RESPONDER: Responder,
```

Comment:
Would the code be that much longer if you didn't use this unstable, generally unknown to people, complicated feature?

Reply:
You could pass them in as functions to the constructor of `MockedExecutor`:
```rust
impl MockedExecutor{
  fn new(store_key_fn, verify_fn) { ... }
}
```
and use in `impl Executor`
```rust
impl Executor for MockedExecutor {
  fn store_key(args) => self.store_key_fn(args)
}
```
and then in your unit test
```rust
let panicking_executor = MockedExecutor::new(store_key_fn_panic(), verify_fn_panic());
```

I understand you like the feature and may feel using it is smart (it is brilliant I give you that) but it's not stable, it's generally unknown and it makes reading, understanding and hence extending the code harder.

Reply:
I can see now how this approach won't work.